packages/foundry/contracts/hooks/CommitMinerHook.sol

```diff
@@ -0,0 +1,119 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "Balancer/BaseHooks.sol";
+import "Balancer/VaultGuard.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+
+contract CommitMinerHook is BaseHooks, VaultGuard, ReentrancyGuard {
+    struct Commit {
+        bytes32 hash;         // The commit hash
+        uint256 blockNumber;  // The block number when the commit was created
+        address pool;         // The pool that generated the commit
+        address swapper;      // Address of the user who performed the swap
+    }
+
+    IERC20 public immutable paymentToken;  // The ERC20 token used for both payments and rewards
+    address private immutable hookDeployer;  // Address of the hook deployer
```

Comment:
This works, but could also derive from Ownable and use `owner()`, which does the same thing in a standardized way, and also enables permissioned functions (e.g., maybe the owner can change the fee percentages later).